src/domain/entities/DataSet.ts

```diff
@@ -41,9 +41,8 @@ export type CoreCompetency = { id: Id; name: string; code: string };
 export type DataSetList = Pick<DataSetAttrs, "id" | "name" | "lastUpdated" | "permissions">;
 
 export class DataSet extends Struct<DataSetAttrs>() {
-    validate(): Either<ValidationError<DataSet>[], DataSet> {
-        const allErrors = this.getValidationErrors();
-        return allErrors.length === 0 ? Either.success(this) : Either.error(allErrors);
+    validate(): ValidationError<DataSet>[] {
```

Comment:
minor, no need to change, just a general note: `ValidationError<DataSet>[]` limits the expansion of the return type: now it's bound to be an array of something. For a more general case: `ValidationErrors<DataSet>`, and then you are free to re-shape the type without changing all the signatures that the object traverses, only the code interacting with its internals.